firebase-perf/src/main/java/com/google/firebase/perf/network/InstrURLConnectionBase.java

```diff
@@ -156,8 +161,13 @@ public long getLastModified() {
 
   public OutputStream getOutputStream() throws IOException {
     try {
-      return new InstrHttpOutputStream(
-          httpUrlConnection.getOutputStream(), networkMetricBuilder, timer);
+      final OutputStream outputStream = httpUrlConnection.getOutputStream();
+      // Make sure we don't pass in a null into InstrHttpOutputStream, since InstrHttpOutputStream
+      // is not null-safe.
+      if (outputStream != null) {
+        return new InstrHttpOutputStream(outputStream, networkMetricBuilder, timer);
+      }
+      return null;
```

Comment:
Is the caller of `getOutputStream()` okay to deal with null?

Reply:
This question would apply to both input and output stream.

I suspect that it depends on whether the developers thinks a null may occur with getInputStream (but since the docs don't mention it can be null, devs may not add a check).

However, we use this class to wrap the original url connect and call the underlying object as a delegate.  We shouldn't be returning our instrumented input stream when the original url connection would have returned a null.  If they did not have Firebase Performance, they would have needed to handle the null case anyways.

Reply:
I could also do `return outputStream` instead of `return null` to showcase that we are not changing the original behavior.